### PR TITLE
refactor(osio): Split osio api client in 3

### LINKED ISSUE #235: Split OsioApiClient in 3
OsioWitClient, OsioJenkinsClient (from JenkinsGitCredentials), and OsioAuthClient following endpoint prefix.

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #235

### CODE REVIEW

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/api/Namespace.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonDeserialize(as = ImmutableNamespace.class)
+@JsonSerialize(as = ImmutableNamespace.class)
+public interface Namespace {
+
```
Comment (gastaldi) [inline addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/api/Namespace.java]: Namespace is used only in Tenant. Why move to a top level class?

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/impl/OsioWitClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    @Override
+    public UserInfo getUserInfo() {
+        final Request userInfoRequest = newAuthorizedRequestBuilder("/api/user").build();
+        return executeAndParseJson(userInfoRequest, OsioWitClientImpl::readUserInfo)
+                .orElseThrow(() -> new BadTenantException("UserInfo not found"));
+    }
+
+    @Override
+    public List<Namespace> getNamespaces() {
```
Comment (gastaldi) [inline addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/impl/OsioWitClientImpl.java]: This method can be private. I don't think we'll be using it outside getTenant

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/web/producers/OsioRequestScopedProducer.java [hunk 3]
```diff
@@ -40,16 +44,33 @@ public OpenShiftService createOpenShiftService(final IdentityProvider identityPr
...
+    public OsioJenkinsClient createOsioJenkinsClient(final HttpServletRequest request,
+                                                     final IdentityProvider identityProvider) {
+        return new OsioJenkinsClientImpl(request.getHeader(HttpHeaders.AUTHORIZATION), identityProvider);
+    }
+
+    @Produces
+    @RequestScoped
+    public Tenant produceTenant(final HttpServletRequest request,
+                                final OsioWitClient osioWitClient) {
+        final String authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
```
Comment (gastaldi) [inline addons/osio-addon/src/main/java/io/fabric8/launcher/osio/web/producers/OsioRequestScopedProducer.java]: Return witClient.getTenant()?

File: base/src/main/java/io/fabric8/launcher/base/http/ExternalRequest.java [hunk 4]
```diff
@@ -27,19 +27,28 @@ private ExternalRequest() {
...
+        try (final Response response = client.newCall(request).execute()) {
             return consumer.apply(response);
         } catch (IOException e) {
             throw new HttpException("Error while executing request", e);
         }
     }
 
+    public static Response execute(Request request) {
+        try (final Response response = client.newCall(request).execute()) {
+            return response;
```
Comment (gastaldi) [inline base/src/main/java/io/fabric8/launcher/base/http/ExternalRequest.java]: -1 Response will be closed.that's why the original method takes a Function

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/web/producers/OsioRequestScopedProducer.java (hunk 3 above)
Comment (ia3andy) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/web/producers/OsioRequestScopedProducer.java]: Oupsy :) go in hollllidaaay haha

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/impl/OsioWitClientImpl.java (hunk 2 above)
Comment (ia3andy) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/impl/OsioWitClientImpl.java]: Yes

File: addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/api/Namespace.java (hunk 1 above)
Comment (ia3andy) [reply addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/api/Namespace.java]: Yes

File: base/src/main/java/io/fabric8/launcher/base/http/ExternalRequest.java (hunk 4 above)
Comment (ia3andy) [reply base/src/main/java/io/fabric8/launcher/base/http/ExternalRequest.java]: ok it makes sense :)

**Pull request merged**